nomad/leader.go

```diff
@@ -1966,12 +1963,13 @@ func (s *Server) getOrCreateSchedulerConfig() *structs.SchedulerConfiguration {
 	return config
 }
 
+var minVersionKeyring = version.Must(version.NewVersion("1.4.0"))
+
 // initializeKeyring creates the first root key if the leader doesn't
 // already have one. The metadata will be replicated via raft and then
 // the followers will get the key material from their own key
 // replication.
-func (s *Server) initializeKeyring() error {
-
+func (s *Server) initializeKeyring(stopCh <-chan struct{}) error {
```

Comment:
This function now gets called in a go routine, so do we want to remove the error return and log any error messages?

Reply:
Good call.

Reply:
Done in [3845b9c](https://github.com/hashicorp/nomad/pull/14821/commits/3845b9cf8b88e934ff4aa524f42d8af1fbf37071)